src/main/java/com/cloudbees/plugins/credentials/impl/UsernamePasswordCredentialsImpl.java

```diff
@@ -53,6 +55,9 @@
     @NonNull
     private final Secret password;
 
+    @Nullable
+    private Boolean usernameSecret = false;
```

Comment:
Default false in the Java constructor which would be used only from tests or scripts.